tiledb/sm/array_schema/domain.cc

```diff
@@ -1200,9 +1112,8 @@ T Domain::floor_to_tile(T value, unsigned dim_idx) const {
   if (tile_extents_ == nullptr)
     return domain[2 * dim_idx];
 
-  return ((value - domain[2 * dim_idx]) / tile_extents[dim_idx]) *
-             tile_extents[dim_idx] +
-         domain[2 * dim_idx];
+  uint64_t div = (value - domain[2 * dim_idx]) / tile_extents[dim_idx];
+  return (T)div * tile_extents[dim_idx] + domain[2 * dim_idx];
```

Comment:
I see this is essentially unchanged, but can this overflow when casting back to `T`?

Reply:
`uint64_t` definitely makes more sense, but we will have to change the signature of the function. For now I don't think that this creates problems.